x-pack/libbeat/common/aws/credentials.go

```diff
@@ -18,11 +20,13 @@ type ConfigAWS struct {
 	ProfileName          string `config:"credential_profile_name"`
 	SharedCredentialFile string `config:"shared_credential_file"`
 	Endpoint             string `config:"endpoint"`
+	RoleArn              string `config:"role_arn"`
 }
 
 // GetAWSCredentials function gets aws credentials from the config.
 // If access_key_id and secret_access_key are given, then use them as credentials.
-// If not, then load from aws config file. If credential_profile_name is not
+// If role_arn is given, assume the IAM role instead.
+// If none of the above is given, then load from aws config file. If credential_profile_name is not
 // given, then load default profile from the aws config file.
 func GetAWSCredentials(config ConfigAWS) (awssdk.Config, error) {
```

Comment:
nice to have in the future - it might be valid to put some prints for debugging the auth mode selection (access key, credential profile, etc.)

Reply:
Good idea! Let me add some in this PR so I don't forget about this!

Reply:
Just added a logger for `GetAWSCredentials` function.